modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -917,17 +947,149 @@ public String addPredecessor(String predecessorDetails){
+                        appendText(doc, predecessorElement, "id", mns, predecessorList.getJSONObject(i).getString("id"));
+                    }
+                    String newProcessContent = xmlToString(doc);
+                    resource.setContent(newProcessContent);
+                    reg.put(processAssetPath, resource);
+                }
+            }
+
+        }catch (Exception e){
+            log.error(e);
+        }
+        return "OK";
+    }
+
+    public String deleteSubprocess(String deleteSubprocess){
+        try{
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+
+            if(registryService != null){
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+
+                JSONObject processInfo = new JSONObject(deleteSubprocess);
+                String processName = processInfo.getString("processName");
+                String processVersion = processInfo.getString("processVersion");
+                JSONObject subprocess = processInfo.getJSONObject("deleteSubprocess");
+
+                String processAssetPath = "processes/" + processName + "/" + processVersion;
+                Resource resource = reg.get(processAssetPath);
+                String processContent = new String((byte[]) resource.getContent());
+                Document doc = stringToXML(processContent);
+
+                if(subprocess != null){
+                    NodeList subprocessElements = ((Element)doc.getFirstChild()).getElementsByTagName("subprocess");
+                    for(int i = 0 ; i < subprocessElements.getLength() ; i++){
+                        Element subprocessElement = (Element) subprocessElements.item(i);
+                        String subprocessName = subprocessElement.getElementsByTagName("name").item(0).getTextContent();
+                        String subprocessPath =
+                                subprocessElement.getElementsByTagName("path").item(0).getTextContent();
+                        String subprocessId =
+                                subprocessElement.getElementsByTagName("id").item(0).getTextContent();
+
+                        if(subprocessName.equals(subprocess.getString("name")) && subprocessPath.equals(subprocess.getString("path")) &&
+                           subprocessId.equals(subprocess.getString("id"))){
+                            subprocessElement.getParentNode().removeChild(subprocessElement);
+                            break;
+                        }
+                    }
+                    String newProcessContent = xmlToString(doc);
+                    resource.setContent(newProcessContent);
+                    reg.put(processAssetPath, resource);
+                }
+            }
+        }catch (Exception e){
+            log.error(e);
+        }
+        return "OK";
+    }
```

Comment:
Because here I used a jaggery file to invoke to the backend code which provided in the ProcessStore and print the response as a String. Now I changed it to the boolean type and cahnge the jag files also. fixed in https://github.com/wso2/product-pc/pull/36